### PR TITLE
feat: compute dbt model lineage across databases

### PR DESCRIPTION
**Pull request by aballiet**
### Describe your changes:

Fixes https://github.com/open-metadata/OpenMetadata/issues/15973

I worked on ... because ...

#
### Type of change:

- [ ] Bug fix
- [x] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: ingestion/src/metadata/ingestion/source/database/dbt/metadata.py [hunk 1]
```diff
@@ -116,6 +117,16 @@ def __init__(self, config: WorkflowSource, metadata: OpenMetadata):
             else "dbtTags"
         )
 
+        # Create a service map to get dbt model lineage across different databases
+        all_databases = self.metadata.es_search_from_fqn(
+            entity_type=Database,
+            fqn_search_string="*",
+        )
+        self.database_services_map = {
```
Comment (OnkarVO7) [inline ingestion/src/metadata/ingestion/source/database/dbt/metadata.py:125]: This map might fail when you will have databases with same names under different services.

**Review by OnkarVO7 — COMMENTED** [review]
@aballiet thanks for the PR. Left a comment where the map might not work correctly.

The approach I would think to get here is, where we build the parent_fqn we can search the elasticsearch using `*.db.schema.table` this way we can search through all the dbs for the table from manifest.json
Let me know your thoughts

File: ingestion/src/metadata/ingestion/source/database/dbt/metadata.py (hunk 1 above)
Comment (aballiet) [reply ingestion/src/metadata/ingestion/source/database/dbt/metadata.py:125]: True ! I did not know it was allowed in backend

File: ingestion/src/metadata/ingestion/source/database/dbt/metadata.py [hunk 2]
```diff
@@ -948,7 +948,7 @@ def add_dbt_test_result(self, dbt_test: dict):
                     test_case_fqn = fqn.build(
                         self.metadata,
                         entity_type=TestCase,
-                        service_name=self.config.serviceName,
+                        service_name="*",
```
Comment (OnkarVO7) [inline ingestion/src/metadata/ingestion/source/database/dbt/metadata.py:951]: Here let's use the `source_elements[0]` instead of `*`

**Review by OnkarVO7 — CHANGES_REQUESTED** [review]

File: ingestion/src/metadata/ingestion/source/database/dbt/metadata.py (hunk 2 above)
Comment (aballiet) [reply ingestion/src/metadata/ingestion/source/database/dbt/metadata.py:951]: Ok

Comment (aballiet) [reply ingestion/src/metadata/ingestion/source/database/dbt/metadata.py]: Should we remove `self.config.serviceName` param as it is not needed anymore

Comment (OnkarVO7) [reply ingestion/src/metadata/ingestion/source/database/dbt/metadata.py]: Yes if it is not used anywhere we can remove it

Comment (aballiet) [reply ingestion/src/metadata/ingestion/source/database/dbt/metadata.py]: Done 👍

**Review by OnkarVO7 — APPROVED** [review]

**Pull request merged**